scripts/missions/asa/03_That_Which_Curdles_Blood.lua

```diff
@@ -0,0 +1,112 @@
+-----------------------------------
+-- A Shantotto Ascension
+-- A Shantotto Ascension M3
+-----------------------------------
+-- !addmission 11 2
+-- Kuroido-Moido : !pos -113.78 -3.25 105.40 240
+-- Faulpie       : !pos -178.88 -1 9.89 230
+-- Abd-al-Raziq  : !pos 126.7 2.0 -0.23 234
+-- Trodden Snow  : !pos -19.7 -17.3 104.4 126
+-----------------------------------
+require('scripts/globals/missions')
+require('scripts/settings/main')
+require('scripts/globals/interaction/mission')
+require('scripts/globals/zone')
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.ASA, xi.mission.id.asa.THAT_WHICH_CURDLES_BLOOD)
+
+local function handleTradeEvent(player, trade, firstId)
+    local asaKit = player:getCharVar("ASA_kit")
+    if npcUtil.tradeHasExactly(trade, asaKit) then
+        return mission:progressEvent(firstId)
+    end
+end
+
+local handleTradeEventFinish = function(player, csid, option, npc)
+    if mission:complete(player) then
+        player:confirmTrade()
+
+        npcUtil.giveKeyItem(player, {
```

Comment:
Neat, didn't know that this worked.  If this is the case, define a mission.reward block, and define this table of KIs there (can see other missions for example of mission.reward table, but in this case instead of a single KI, you'd put the table of KIs there)

Also, could you move the mission.reward section above local functions?  Just want to be consistent with other missions and quests